### PR TITLE
API Changes: Account (User is being moved to Account with some updates)

### PR DESCRIPTION
**Pull request by sameerag**
> There is one pending change for this PR - accountIdentifier: requirements finalization in progress with the server.

This PR consists of introducing Account object (previously known as User)

Changes made:
- Account.ts - new parameters introduced standardizing the object and compatibility with other MSALs
- getUser() API now changes to getAccount()
- All the user references in the library shifted to account. 
- All dependencies in other .ts files addressed

After this migration any reference to the "User" in the documentation/comments would mean the developer using the library. The logged in context is referred to as "account" as it is possible for a single user to have multiple accounts and hence more realistic nomenclature.

### CODE REVIEW

File: lib/msal-core/src/Account.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ *  All Rights Reserved
+ *  MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
```
[the remark this answers is not in the document]
Comment (sameerag) [reply lib/msal-core/src/Account.ts]: Yes. I will do this once I merge #621 to integration branch and pull dev into the integration branch.

File: lib/msal-core/src/Account.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) Microsoft Corporation
```
Comment (DarylThayil) [inline lib/msal-core/src/Account.ts]: Use new copyright in all the other files

File: lib/msal-core/src/Account.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    /**
+     * @hidden
+     * @param idToken
+     * @param clientInfo
+     */
+    static createAccount(idToken: IdToken, clientInfo: ClientInfo): Account {
+
+        // create accountIdentifier
+        // TODO: should we advocate idToken.oid+idToken.tid for all MSALs??
+        let accountIdentifier: string;
```
Comment (DarylThayil) [inline lib/msal-core/src/Account.ts]:
```js
const accountIdentifier: string = idToken.objectId ||  idToken.subject;
```

File: lib/msal-core/src/Account.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+        if (idToken.objectId) {
+          accountIdentifier = idToken.objectId;
+        }
+        else {
+          accountIdentifier = idToken.subject;
+        }
+
+        // create homeAccountIdentifier
+        let uid: string;
```
Comment (DarylThayil) [inline lib/msal-core/src/Account.ts]: const uid: string = clientInfo  ? clientInfo.uid : '';
const utid: string = clientInfo ? clientInfo.utid: ''

File: lib/msal-core/src/Account.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        let utid: string;
+
+        if (!clientInfo) {
+            uid = "";
```
Comment (DarylThayil) [inline lib/msal-core/src/Account.ts]: we should stick to single quotes, it is convention in js community

File: lib/msal-core/src/Storage.ts [hunk 6]
```diff
@@ -179,15 +179,18 @@ export class Storage {// Singleton
     }
 
     /**
-     * Create acquireTokenUserKey to cache user object
+     * Create acquireTokenAccountKey to cache account object
+     * @param accountId
+     * @param state
      */
-    static generateAcquireTokenUserKey(userId: any, state: string): string {
+    static generateAcquireTokenAccountKey(accountId: any, state: string): string {
```
Comment (DarylThayil) [inline lib/msal-core/src/Storage.ts]: should this have a type

File: lib/msal-core/src/Account.ts (hunk 2 above)
Comment (sameerag) [reply lib/msal-core/src/Account.ts]: Yes. As I mentioned above, I plan to rebase the final integration branch to dev soon and this should be covered in the same.

File: lib/msal-core/src/Account.ts (hunk 3 above)
Comment (sameerag) [reply lib/msal-core/src/Account.ts]: Was keeping the code pattern constant. Will change this up soon.

File: lib/msal-core/src/Account.ts (hunk 5 above)
Comment (sameerag) [reply lib/msal-core/src/Account.ts]: Sure.

File: lib/msal-core/src/Storage.ts (hunk 6 above)
Comment (sameerag) [reply lib/msal-core/src/Storage.ts]: no. It is not supposed to as it is a key off with no structure reference from what I understand. I can check again.

File: lib/msal-core/src/Account.ts (hunk 4 above)
Comment (sameerag) [reply lib/msal-core/src/Account.ts]: Done.

**Pull request merged**